lib/receiver/receiver_impl.cc

```diff
@@ -192,7 +192,14 @@ namespace gr
         sch_search_handler(input, noutput_items);
         break;
       case synchronized:
-        synchronized_handler(input, input_items, noutput_items);
+	synchronized_handler(input, input_items, noutput_items,0);
+	size_t inputs_to_process = d_cell_allocation.size();
+	if (d_process_uplink)
+        	inputs_to_process *= 2;
+	for(int input_nr=0;input_nr<inputs_to_process;input_nr++){
+		boost::thread t1(&receiver_impl::synchronized_handler,this,input, input_items, noutput_items, input_nr);
```

Comment:
It could be dangerous to pass the same `this` pointer to all threads.  The `synchronized_handler` is not a const method and it modifies the object members. When multiple threads running in parallel access the same object they could modify its members which could leads to an undefined behavior.